### PR TITLE
Added Self where possible in LinkedList (updated)

### PR DESCRIPTION
**Pull request by cmlsharp**
Updated pull request

@Gankro

### CODE REVIEW

File: src/libcollections/linked_list.rs [hunk 1]
```diff
@@ -898,9 +898,9 @@ impl<T> IntoIterator for LinkedList<T> {
 #[stable(feature = "rust1", since = "1.0.0")]
 impl<'a, T> IntoIterator for &'a LinkedList<T> {
     type Item = &'a T;
-    type IntoIter = Iter<'a, T>;
+    type IntoIter = Iter<'a, Self::IntoIter>;
```
Comment (Gankro) [inline src/libcollections/linked_list.rs]: This should have still been T, and is causing infinite recursion in type checking.

Comment (crossroads1112) [inline src/libcollections/linked_list.rs]: Ahh, I meant to type `Self::Item`

**Comment by Gankro** [comment]
```
rustc: x86_64-unknown-linux-gnu/stage0/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcollections
src/libcollections/linked_list.rs:899:1: 906:2 error: type mismatch resolving `<linked_list::Iter<'_, &T> as core::iter::Iterator>::Item == &T`:
 expected &-ptr,
    found type parameter [E0271]
src/libcollections/linked_list.rs:899 impl<'a, T> IntoIterator for &'a LinkedList<T> {
src/libcollections/linked_list.rs:900     type Item = &'a T;
src/libcollections/linked_list.rs:901     type IntoIter = Iter<'a, Self::Item>;
src/libcollections/linked_list.rs:902 
src/libcollections/linked_list.rs:903     fn into_iter(self) -> Self::IntoIter {
src/libcollections/linked_list.rs:904         self.iter()
                                      ...
src/libcollections/linked_list.rs:899:1: 906:2 help: run `rustc --explain E0271` to see a detailed explanation
src/libcollections/linked_list.rs:899:1: 906:2 note: required by `core::iter::IntoIterator`
error: aborting due to previous error
make: *** [x86_64-unknown-linux-gnu/stage0/lib/rustlib/x86_64-unknown-linux-gnu/lib/stamp.collections] Error 101
```

**Comment by Gankro** [comment]
`Iter<'a, T>`, not `Iter<'a, &T>`

**Comment by oli-obk** [comment]
ping @Gankro , this has been updated, but github doesn't notify

**Comment by bluss** [comment]
Thanks, this looks good. Can you please squash together all the commits?

A reviewer can r=me when the squash is done.

**Comment by bluss** [comment]
Like oli-obk said, there's no notification for new commits or pushes, so a message in the PR thread is good way to notify those subscribed to the PR.

**Comment by cyplo** [comment]
Hello ! Is this one still relevant ? If so - is there anything blocking it from getting merged ? thanks !

**Comment by mrhota** [comment]
@crossroads1112 if you're short on time, I can re-open this into another PR and squash these commits for you. Let me know.

**Comment by bluss** [comment]
Thanks so much @crossroads1112, I will close this PR for inactivity, but you (or anyone else) are welcome to submit this change again.

**Pull request closed without merging**